### PR TITLE
security: update some deps

### LINKED ISSUE #3613: 10 vulnerabilities (3 high, 7 critical)
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

npm shows `@tinacms/cli` package has 10 vulnerabilities. Here is the full log:
```
# npm audit report

trim  <0.0.3
Severity: high
Regular Expression Denial of Service in trim - https://github.com/advisories/GHSA-w5p7-h5w8-2hfq
fix available via `npm audit fix --force`
Will install @tinacms/cli@1.0.3, which is a breaking change
node_modules/trim
  remark-parse  <=8.0.3
  Depends on vulnerable versions of trim
  node_modules/@textlint/markdown-to-ast/node_modules/remark-parse
    @textlint/markdown-to-ast  6.0.8 - 6.3.5
    Depends on vulnerable versions of remark-parse
    node_modules/@textlint/markdown-to-ast
      doctoc  1.3.0 - 2.1.0
      Depends on vulnerable versions of @textlint/markdown-to-ast
      Depends on vulnerable versions of underscore
      node_modules/doctoc
        concat-md  *
        Depends on vulnerable versions of doctoc
        node_modules/concat-md
          @tinacms/mdx  <=0.0.0-20230214023507 || >=1.0.3
          Depends on vulnerable versions of concat-md
          node_modules/@tinacms/mdx
            @tinacms/graphql  >=1.0.3
            Depends on vulnerable versions of @tinacms/mdx
            node_modules/@tinacms/graphql
              @tinacms/cli  <=0.0.0-20230214023507 || >=1.0.4
              Depends on vulnerable versions of @tinacms/datalayer
              Depends on vulnerable versions of @tinacms/graphql
              node_modules/@tinacms/cli
              @tinacms/datalayer  >=1.1.0
              Depends on vulnerable versions of @tinacms/graphql
              node_modules/@tinacms/datalayer

underscore  1.3.2 - 1.12.0
Severity: critical
Arbitrary Code Execution in underscore - https://github.com/advisories/GHSA-cf4h-3jhx-xvhq
fix available via `npm audit fix --force`
Will install @tinacms/cli@1.0.3, which is a breaking change
node_modules/underscore

10 vulnerabilities (3 high, 7 critical)

To address all issues (including breaking changes), run:
  npm audit fix --force
```

My package.json:

```
{
  "name": "@example/basics",
  "type": "module",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "netlify:dev": "netlify dev",
    "dev": "tinacms dev -c \"astro dev --host\"",
    "start": "tinacms dev -c \"astro dev --host\"",
    "build": "tinacms build && astro build",
    "preview": "astro preview",
    "astro": "astro"
  },
  "dependencies": {
    "@astrojs/prefetch": "^0.1.2",
    "@astrojs/sitemap": "^1.0.1",
    "@fortawesome/fontawesome-pro": "^6.3.0",
    "@tinacms/auth": "^1.0.3",
    "@tinacms/cli": "^1.2.0",
    "astro": "^2.0.11",
    "astro-compress": "^1.1.32",
    "bootstrap": "^5.2.3",
    "dayjs": "^1.11.7",
    "next-tinacms-s3": "^1.1.1",
    "serverless-http": "^3.1.1",
    "tinacms": "^1.2.1"
  },
  "devDependencies": {
    "@fullhuman/postcss-purgecss": "^5.0.0",
    "@types/bootstrap": "^5.2.6",
    "mdast-util-to-string": "^3.1.1",
    "prettier": "^2.8.4",
    "prettier-plugin-astro": "^0.8.0",
    "reading-time": "^1.5.0",
    "sass": "^1.58.1",
    "terser": "^5.16.3",
    "unist-util-visit-parents": "^5.1.3"
  }
}
```

### Reproduction

npm install @tinacms/cli

### Steps to reproduce

_No response_

### System Info

```shell
System:
  OS: macOS 13.1
  CPU: (10) arm64 Apple M1 Pro
  Memory: 113.69 MB / 16.00 GB
  Shell: 5.8.1 - /bin/zsh
Binaries:
  Node: 16.18.0 - /opt/homebrew/opt/node@16/bin/node
  Yarn: 1.22.19 - ~/upwork/nick/cashforcars.io-next/node_modules/.bin/yarn
  npm: 8.19.2 - /opt/homebrew/opt/node@16/bin/npm
Browsers:
  Chrome: 110.0.5481.77
  Firefox: 104.0.2
  Safari: 16.2
npmPackages:
  @tinacms/auth: ^1.0.3 => 1.0.3 
  @tinacms/cli: ^1.2.0 => 1.2.0 
  tinacms: ^1.2.1 => 1.2.1
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

### PR DESCRIPTION
**Pull request by jamespohalloran**
closes #3613
closes ENG-821

### CODE REVIEW

File: packages/@tinacms/datalayer/package.json [hunk 1]
```diff
@@ -28,6 +28,26 @@
...
+    "fast-glob": "^3.2.5",
+    "fs-extra": "^9.0.1",
+    "glob-parent": "^6.0.2",
+    "graphql": "15.8.0",
+    "gray-matter": "^4.0.2",
+    "isomorphic-git": "^1.10.3",
+    "js-yaml": "^3.14.0",
+    "jsonpath-plus": "^6.0.1",
+    "level": "^7.0.1",
+    "leveldown": "^6.1.0",
```
Comment (kldavis4) [inline packages/@tinacms/datalayer/package.json:44]: @jamespohalloran what's up with all these deps getting added?

Comment (jamespohalloran) [reply packages/@tinacms/datalayer/package.json:44]: Good question, not sure how that happened, taking a look

Comment (kldavis4) [reply packages/@tinacms/datalayer/package.json:44]: Some of those look like the pre-abstractlevel dependencies

Comment (jamespohalloran) [reply packages/@tinacms/datalayer/package.json]: I think there may have been a funky merge. Just removed those

**Review by kldavis4 — APPROVED** [review]

**Pull request merged**